src/Chapter2.hs

```diff
@@ -621,8 +627,10 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
-
+takeEven :: [a] -> [a]
+takeEven [] = []
+takeEven (x:_:xs) = x : takeEven xs -- base pattern
+takeEven (x:_)    = x : takeEven []
```

Comment:
This case is actually is needed to cover `[x]` case, as anything else will pattern match earlier. 🙂 
So, you can avoid additional function call:

```suggestion
takeEven [x]    = [x]
```